src/languages/r.js

```diff
@@ -131,48 +146,91 @@ export default function(hljs) {
           {begin: "'", end: "'", relevance: 0}
         ],
       },
+
+      // Matching numbers immediately following punctuation and operators is
+      // tricky since we need to look at the character ahead of a number to
+      // ensure the number is not part of an identifier, and we cannot use
+      // negative look-behind assertions. So instead we explicitly handle all
+      // possible combinations of (operator|punctuation), number.
+      // TODO: replace with negative look-behind when available
+      // { begin: /(?<![a-zA-Z0-9._])0[xX][0-9a-fA-F]+\.[0-9a-fA-F]*[pP][+-]?\d+i?/ },
+      // { begin: /(?<![a-zA-Z0-9._])0[xX][0-9a-fA-F]+([pP][+-]?\d+)?[Li]?/ },
+      // { begin: /(?<![a-zA-Z0-9._])(\d+(\.\d*)?|\.\d+)([eE][+-]?\d+)?[Li]?/ }
       {
         relevance: 0,
-        className: {
-          2: "number"
-        },
         variants: [
-          // TODO: replace with negative look-behind when available
-          // { begin: /(?<![a-zA-Z0-9._])0[xX][0-9a-fA-F]+\.[0-9a-fA-F]*[pP][+-]?\d+i?/ },
-          // { begin: /(?<![a-zA-Z0-9._])0[xX][0-9a-fA-F]+([pP][+-]?\d+)?[Li]?/ },
-          // { begin: /(?<![a-zA-Z0-9._])(\d+(\.\d*)?|\.\d+)([eE][+-]?\d+)?[Li]?/ }
           {
-            // Special case: only hexadecimal binary powers can contain fractions.
+            scope: {
+              1: 'operator',
+              2: 'number'
+            },
             match: [
-              /[^a-zA-Z0-9._]/, // not part of an identifier
-              /0[xX][0-9a-fA-F]+\.[0-9a-fA-F]*[pP][+-]?\d+i?/
+              OPERATORS_RE,
+              NUMBER_TYPES_RE
             ]
           },
           {
+            scope: {
+              1: 'operator',
+              2: 'number'
+            },
             match: [
-              /[^a-zA-Z0-9._]/, // not part of an identifier
-              /0[xX][0-9a-fA-F]+(?:[pP][+-]?\d+)?[Li]?/
+              /%[^%]*%/,
```

Comment:
What is this again in R?  And %% is also valid?

Reply:
Any `%…%` sequence is a valid binary operator, `%%` included.